src/app/pages/Project.tsx

```diff
@@ -1,51 +1,98 @@
 import React, {useState, useEffect} from 'react';
 import {
+  fetchProject,
   createDocument,
   deleteDocument,
   fetchDocumentsProject,
+  fetchDocumentsAndMultiPacksProject,
+  createCrossDoc,
+  deleteCrossDoc,
 } from '../lib/api';
 import {Link, useHistory} from 'react-router-dom';
 import DropUpload from '../components/dropUpload';
 import {FileWithPath} from 'react-dropzone';
 
 function Docs() {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [projectInfo, setProjectInfo] = useState<any>();
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const [docs, setDocs] = useState<any[]>([]);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [crossdocs, setCrossDocs] = useState<any[]>();
 
   const history = useHistory();
 
   useEffect(() => {
-    updateDocs().catch(() => {
+    getProjectInfo().catch(() => {
       history.push('/login');
     });
   }, [history]);
 
-  function updateDocs() {
-    const project_id = window.location.pathname.split('/').pop()!;
+  useEffect(() => {
+    updateDocs();
+  }, [projectInfo]);
 
-    return fetchDocumentsProject(project_id).then(docs => {
-      setDocs(docs);
+  function getProjectInfo() {
+    const project_id = window.location.pathname.split('/').pop()!;
+    return fetchProject(project_id).then(info => {
+      setProjectInfo(info);
     });
   }
-
-  function handleAdd(filesToUpload: FileWithPath[]) {
-    const project_id = window.location.pathname.split('/').pop()!;
-
-    filesToUpload.forEach(f => {
-      const reader = new FileReader();
-      reader.readAsText(f);
-      reader.onload = function () {
-        createDocument(f.name, reader.result as string, project_id).then(() => {
-          updateDocs();
+  function updateDocs() {
+    if (projectInfo) {
+      const project_id = window.location.pathname.split('/').pop()!;
+      if (projectInfo.project_type === 'indoc') {
+        return fetchDocumentsProject(project_id).then(docs => {
+          setDocs(docs);
         });
-      };
-    });
+      } else if (projectInfo.project_type === 'crossdoc') {
+        return fetchDocumentsAndMultiPacksProject(project_id).then(result => {
+          console.log(result);
+          setDocs(result.docs);
+          setCrossDocs(result.crossdocs);
+        });
+      }
+    }
+  }
+  function handleAdd(filesToUpload: FileWithPath[], pack_type = 'single_pack') {
+    const project_id = window.location.pathname.split('/').pop()!;
```

Comment:
I see this is used a number of time, maybe it's good idea to define it outside.

Reply:
Updated in newer commit.